tests/scanner/scanners/service_account_key_test.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service Account Key Scanner Tests."""
+
+import unittest
+import mock
+from datetime import datetime, timedelta
+
+from tests import unittest_utils
+from tests.services.util.db import create_test_engine
+from google.cloud.forseti.scanner.scanners import service_account_key_scanner
+from google.cloud.forseti.common.util.string_formats import DEFAULT_FORSETI_TIMESTAMP
+from google.cloud.forseti.services.dao import ModelManager
+
+"""
+Assumptions: In data/service_account_key_test_rules.yaml, max_age is set to 
+100 days.
+
+Test: Create a service account with two keys, one 99 days old, one 101 days old.
+The 101 day old key should be flagged as a violation but not the 99 day old.
+"""
+
+SERVICE_ACCOUNT_ID = '123456789012345678901'
+SERVICE_ACCOUNT_NAME = 'projects/foo/serviceAccounts/test-service-account/test-service-account@developer.gserviceaccount.com'
+SERVICE_ACCOUNT_DATA = """
+{"displayName": "Service Account Key Age Test Service Account",
+ "email": "test-service-account@developer.gserviceaccount.com",
+ "name": "%s",
+ "oauth2ClientId": "%s",
+ "projectId": "foo",
+ "uniqueId": "%s"}
+""" % (SERVICE_ACCOUNT_NAME, SERVICE_ACCOUNT_ID, SERVICE_ACCOUNT_ID)
+
+TIME_NOW = datetime.utcnow()
+
+UN_ROTATED_KEY_AGE_DAYS = 101
```

Comment:
If you did the naming change above, this should also be changed:
```
KEY_AGE_MORE_THAN_MAX_AGE = 101
KEY_DATETIME_MORE_THAN_MAX_AGE = ...
KEY_TIME_MORE_THAN_MAX_AGE = ...
```

Reply:
Yes, that makes more sense. Changed!